File: python/valid_parentheses.py
def is_valid(str):
    if len(str) % 2 != 0:
        return False
    par_dict = {'(': ')', '[': ']', '{': '}'}
    stack = []
    for char in str:
        if char in par_dict.keys():
            stack.append(char)
        else:
            if stack == []:
                return False
            open_brac = stack.pop()
            if char != par_dict[open_brac]:
                return False
    return stack == []

File: go/valid_parentheses.go
package main

import "github.com/golang-collections/collections/stack"

func isValid(s string) bool {
	stack := stack.New()

	for _, char := range s {
		if char == '(' {
			stack.Push(')')
		} else if char == '[' {
			stack.Push(']')
		} else if char == '{' {
			stack.Push('}')
		} else if char != stack.Pop() {
			break
		}
	}
	return stack.Len() == 0
}
